src/ui/components/CommandPalette/CommandPalette.tsx

```diff
@@ -1,38 +1,118 @@
 import { connect, ConnectedProps } from "react-redux";
-import { actions } from "ui/actions";
+import { actions, UIStore } from "ui/actions";
+import hooks from "ui/hooks";
+import { selectors } from "ui/reducers";
+import { UIState } from "ui/state";
+import { UserSettings } from "ui/types";
 import { TextInput } from "../shared/Forms";
 import Modal from "../shared/NewModal";
 import CommandButton from "./CommandButton";
 const { filter } = require("fuzzaldrin-plus");
 
-export type Command = { key: CommandKey; label: string; shortcut?: string };
+export type Command = {
+  key: CommandKey;
+  label: string;
+  shortcut?: string;
+  settingKey?: keyof UserSettings;
+};
 export type CommandKey =
-  | "open_viewer"
+  | "open_console"
   | "open_devtools"
+  | "open_elements"
+  | "open_network_monitor"
+  | "open_viewer"
+  | "open_file_search"
+  | "open_function_search"
   | "open_full_text_search"
-  | "open_sources"
   | "open_outline"
   | "open_print_statements"
-  | "open_console"
+  | "open_react_devtools"
+  | "open_sources"
+  | "show_comments"
+  | "show_console_filters"
+  | "show_events"
   | "show_privacy"
+  | "show_replay_info"
   | "show_sharing";
 
 const COMMANDS: Command[] = [
-  { key: "open_viewer", label: "Open Viewer" },
+  { key: "open_console", label: "Open Console" },
   { key: "open_devtools", label: "Open DevTools" },
-  { key: "open_full_text_search", label: "Open Full Text Search", shortcut: "CmdOrCtrl+Shift+F" },
-  { key: "open_sources", label: "Open Sources" },
+  { key: "open_elements", label: "Open Elements", settingKey: "showElements" },
+  {
+    key: "open_network_monitor",
+    label: "Open Network Monitor",
+    settingKey: "enableNetworkMonitor",
+  },
+  { key: "open_viewer", label: "Open Viewer" },
+  { key: "open_file_search", label: "Search for file", shortcut: "CmdOrCtrl+P" },
+  { key: "open_full_text_search", label: "Search full text", shortcut: "CmdOrCtrl+Shift+F" },
+  { key: "open_function_search", label: "Search for a function", shortcut: "CmdOrCtrl+Shift+P" },
   { key: "open_outline", label: "Open Outline" },
   { key: "open_print_statements", label: "Open Print Statements" },
-  { key: "open_console", label: "Open Console" },
-  { key: "show_privacy", label: "Show Privacy Information" },
+  { key: "open_react_devtools", label: "Open React DevTools", settingKey: "showReact" },
+  { key: "open_sources", label: "Open Sources" },
+  { key: "show_comments", label: "Show Comments" },
+  { key: "show_console_filters", label: "Show Console Filters" },
+  { key: "show_events", label: "Show Events" },
+  { key: "show_privacy", label: "Show Privacy" },
+  { key: "show_replay_info", label: "Show Replay Info" },
   { key: "show_sharing", label: "Show Sharing Options" },
 ];
 
-function CommandPalette({ hideCommandPalette, executeCommand }: PropsFromRedux) {
+const DEFAULT_COMMANDS: CommandKey[] = [
+  "open_file_search",
+  "open_function_search",
+  "open_full_text_search",
+  "open_sources",
+];
```

Comment:
love bumping these to the top for the default, but I would still show the others below. Being able to scroll through the full list is a nice feature

Reply:
In fact, keeping these top results on the top is important so that there isn't as big of a jump when you start searching.

Reply:
I misgivings about putting the whole list there, because it's important that it feel curated and high importance in its default state. Otherwise you run into [Hick's Law](https://lawsofux.com/hicks-law/), where more options slows people down and overwhelms them.

That said, I think it's great to have the whole list appear anywhere else you hit command-K. That's the moment to go big on all our amazing options. We could even say when you press command-K here, the whole list appears.

Reply:
Are you referring to when it's open by default in the empty editor? I can see how that would be a bit different than the typical command palette experience where you want to optimize for discoverability. See [linear](https://blog.superhuman.com/how-to-build-a-remarkable-command-palette/)

I think a good way to have it both ways is to add some visual separation between the top 4 results and the others. This way you signal, we think these options are great, the others are nice too. This could work in both contexts and keep us consistent

Reply:
I could go for that -- but would the second list be *all* options or another group of cool things?

I think all would be a bit much. But we could try it.

Reply:
Actually, on second thought ...

I don't really mind since we're cutting things off after the first four. My concern was if the region stretched forever, but my concern goes away if we crop. Never mind!